agent/engine/execcmd/manager_init_task_linux.go

```diff
@@ -187,6 +168,21 @@ func buildContainerNameForBinary(c *apicontainer.Container) string {
 	return cn
 }
 
+// AddAgentConfigMount adds the ExecAgentConfigFile to the hostConfig binds
+func (m *manager) AddAgentConfigMount(hostConfig *dockercontainer.HostConfig, execMD apicontainer.ExecCommandAgentMetadata) error {
+	sessionLimit := execMD.SessionLimit
+	if sessionLimit <= 0 {
+		sessionLimit = defaultSessionLimit
+	}
+	configFile, err := GetExecAgentConfigFileName(sessionLimit)
+	if err != nil {
+		return fmt.Errorf("could not generate ExecAgent Config File: %v", err)
+	}
+	configBindMount := filepath.Join(HostExecConfigDir, configFile) + ":" + ContainerConfigFile + ":ro"
+	hostConfig.Binds = append(hostConfig.Binds, configBindMount)
+	return nil
+}
+
 var GetExecAgentConfigFileName = getAgentConfigFileName
```

Comment:
This probably should not be exported

Reply:
We use this method in engine package for integration test, hence the export.